core/snapshot_creator.rs

```diff
@@ -0,0 +1,59 @@
+// Note: This is a nearly identical rewrite of core/libdeno/snapshot_creator.cc
+// but in Rust.
+//
+// This snapshot program is considered "basic" because the code being
+// snapshotted cannot call ops.
+
+#![allow(dead_code)]
+
+// TODO(ry) DRY up this repeated code from lib.rs
+
+#[macro_use]
+extern crate log;
+extern crate futures;
+extern crate libc;
+
+mod flags;
+mod isolate;
+mod js_errors;
+mod libdeno;
+mod module_specifier;
+mod modules;
+mod shared_queue;
+
+use crate::isolate::js_check;
+use crate::isolate::Isolate;
+use crate::isolate::StartupData;
+use std::env;
+use std::io::Write;
+
+fn main() {
+  let args: Vec<String> = env::args().collect();
+  // NOTE: `--help` arg will display V8 help and exit
+  let args = flags::v8_set_flags(args);
+
+  let (snapshot_out_bin, js_filename) = if args.len() == 3 {
+    (args[1].clone(), args[2].clone())
+  } else {
+    eprintln!("Usage: snapshot_creator <out_dir> <js_filename>");
+    std::process::exit(1);
+  };
+
+  let js_source =
+    std::fs::read(&js_filename).expect("couldn't read js_filename");
+  let js_source_str = std::str::from_utf8(&js_source).unwrap();
+
+  let will_snapshot = true;
+  let mut isolate = Isolate::new(StartupData::None, will_snapshot);
+
+  js_check(isolate.execute(&js_filename, js_source_str));
```

Comment:
No biggie, but I really don't understand what the purpose of js_check() is when you have expect() and unwrap().

Reply:
Neither expect nor unwrap will actually display the exception tho...

Reply:
unwrap does it for me...

Reply:
I'm sure unwrap could be made to work by implementing Display correctly (or some other trait) but here's what it looks like.

I've created a small file `foo.js` which contains `throw Error("balh")`.

If I use `unwrap()` instead of `js_check()` the error looks like this:
```
> ./target/debug/snapshot_creator out.bin foo.js
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: JSError { message: "Uncaught Error: balh", source_line: Some(""), script_resource_name: None, line_number: Some(0), start_position: Some(-1), end_position: Some(-1), error_level: Some(8), start_column: Some(-1), end_column: Some(-1), frames: [StackFrame { line: 0, column: 6, script_name: "foo.js", function_name: "", is_eval: false, is_constructor: false, is_wasm: false }] }', src/libcore/result.rs:999:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

If I use `js_check()` the error looks like this
```
> ./target/debug/snapshot_creator out.bin foo.js
thread 'main' panicked at 'Uncaught Error: balh
    at foo.js:1:7', core/isolate.rs:621:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```